crates/block-explorers/src/lib.rs

```diff
@@ -356,44 +281,33 @@ pub struct ClientBuilder {
 // === impl ClientBuilder ===
 
 impl ClientBuilder {
-    /// Configures the etherscan url and api url for the given chain
+    /// Configures the Etherscan url and api url for the given chain
     ///
-    /// Note: This method also sets the chain_id for etherscan multichain verification: <https://docs.etherscan.io/contract-verification/multichain-verification>
+    /// Note: This method also sets the chain_id for Etherscan multichain verification: <https://docs.etherscan.io/contract-verification/multichain-verification>
     ///
     /// # Errors
     ///
-    /// Fails if the chain is not supported by etherscan
+    /// Fails if the chain is not supported by Etherscan
     pub fn chain(self, chain: Chain) -> Result<Self> {
         fn urls(
             (api, url): (impl IntoUrl, impl IntoUrl),
         ) -> (reqwest::Result<Url>, reqwest::Result<Url>) {
             (api.into_url(), url.into_url())
         }
-        let (default_etherscan_api_url, etherscan_url) = chain
+        let (_, etherscan_url) = chain
             .named()
             .ok_or_else(|| EtherscanError::ChainNotSupported(chain))?
             .etherscan_urls()
             .map(urls)
             .ok_or_else(|| EtherscanError::ChainNotSupported(chain))?;
 
-        // V2 etherscan default API urls are different – this handles that case.
-        let etherscan_api_url = if self.etherscan_api_version == EtherscanApiVersion::V2 {
-            Url::parse(ETHERSCAN_V2_API_BASE_URL)
-                .map_err(|_| EtherscanError::Builder("Bad URL Parse".into()))?
-        } else {
-            default_etherscan_api_url?
-        };
+        let etherscan_api_url = Url::parse(ETHERSCAN_V2_API_BASE_URL)
+            .map_err(|_| EtherscanError::Builder("Bad URL Parse".into()))?;
 
         self.with_chain_id(chain).with_api_url(etherscan_api_url)?.with_url(etherscan_url?)
```

Comment:
we should rely on `default_etherscan_api_url` here now rather than manually constructing given `alloy-chains` API urls now contain the `chainid=`

Reply:
fixing this in https://github.com/alloy-rs/chains/pull/196